src/test/java/technology/dice/dicefairlink/iterators/RandomisedCyclicIteratorTest.java

```diff
@@ -0,0 +1,184 @@
+
+    Assertions.assertThat(cyclicIterator).isNotNull();
+
+    assertThat(cyclicIterator.hasNext()).isEqualTo(false);
+    cyclicIterator.next(); // final step throws
+  }
+
+  @Test(expected = NoSuchElementException.class)
+  public void canCallOfFromEmptySet() {
+    RandomisedCyclicIterator cyclicIterator = RandomisedCyclicIterator.of(Collections.EMPTY_SET);
+
+    Assertions.assertThat(cyclicIterator).isNotNull();
+
+    assertThat(cyclicIterator.hasNext()).isEqualTo(false);
+    cyclicIterator.next(); // final step throws
+  }
+
+  @Test
+  public void canOperateWithListOfSingleElement() {
+    final String singleElement = ELEMENT_PREFIX + "_1";
+
+    final List<String> listOfStrings = new LinkedList<>();
+    listOfStrings.add(singleElement);
+
+    RandomisedCyclicIterator<String> cyclicIterator
+        = RandomisedCyclicIterator.<String>of(listOfStrings);
+
+    assertThat(cyclicIterator).isNotNull();
+    for (int cycle = 0; cycle < numberOfElementsToTest; cycle++) {
+      assertThat(cyclicIterator.hasNext()).isEqualTo(true);
+      assertThat(cyclicIterator.next()).isEqualTo(singleElement);
+    }
+  }
+
+  @Test
+  public void canOperateWithSetOfSingleElement() {
+    final String singleElement = ELEMENT_PREFIX + "_1";
+
+    Set<String> setOfString = new LinkedHashSet<>();
+    setOfString.add(singleElement);
+
+    RandomisedCyclicIterator<String> cyclicIterator
+        = RandomisedCyclicIterator.<String>of(setOfString);
+
+    assertThat(cyclicIterator).isNotNull();
+    for (int cycle = 0; cycle < numberOfElementsToTest; cycle++) {
+      assertThat(cyclicIterator.hasNext()).isEqualTo(true);
+      assertThat(cyclicIterator.next()).isEqualTo(singleElement);
+    }
+  }
+
+  @Test
+  public void canOperateWithListOfMultipleIdenticalElements() {
+    final String singleElement = ELEMENT_PREFIX + "_1";
+    final List<String> listOfStrings = new LinkedList<>();
+
+    for (int i = 0; i < numberOfElementsToTest; i++) {
+      listOfStrings.add(singleElement);
+    }
+
+    RandomisedCyclicIterator<String> cyclicIterator
+        = RandomisedCyclicIterator.<String>of(listOfStrings);
+
+    assertThat(cyclicIterator).isNotNull();
+    for (int cycle = 0; cycle < numberOfElementsToTest; cycle++) {
+      for (int i = 0; i < numberOfElementsToTest; i++) {
+        assertThat(cyclicIterator.hasNext()).isEqualTo(true);
+        assertThat(cyclicIterator.next()).isEqualTo(singleElement);
+      }
+    }
+  }
+
+  @Test
+  public void canCompareInetrnalCollectionToExternal_overSameCollectionShuffeled() {
+    List<String> originalElements = new ArrayList<>(numberOfElementsToTest);
+    for (int i = 1; i <= numberOfElementsToTest; i++) {
+      originalElements.add(ELEMENT_PREFIX + i);
+    }
+
+    RandomisedCyclicIterator<String> cyclicIterator
+        = RandomisedCyclicIterator.<String>of(originalElements);
+
+    assertThat(cyclicIterator.hasSameContent(originalElements)).isEqualTo(true);
```

Comment:
Can use `isTrue()`  rather than `isEqualTo(true)` (and `isFalse()` elsewhere)

Reply:
Fixed.